modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java

```diff
@@ -749,25 +749,19 @@ public ModelsMap postProcessModels(ModelsMap objs) {
             boolean addedValidator = false;
             CodegenModel model = m.getModel();
 
-            List<CodegenProperty> inheritedProperties = new ArrayList<>();
+
+            List<CodegenProperty> codegenProperties = new ArrayList<>();
+            codegenProperties.addAll(model.vars);
             if (model.getComposedSchemas() != null) {
                 if (model.getComposedSchemas().getAnyOf() != null) {
-                    inheritedProperties.addAll(model.getComposedSchemas().getAnyOf());
+                    codegenProperties.addAll(model.getComposedSchemas().getAnyOf());
                 }
                 if (model.getComposedSchemas().getOneOf() != null) {
-                    inheritedProperties.addAll(model.getComposedSchemas().getOneOf());
+                    codegenProperties.addAll(model.getComposedSchemas().getOneOf());
+                }
+                if (model.getComposedSchemas().getAllOf() != null) {
+                    codegenProperties.addAll(model.getComposedSchemas().getAllOf());
                 }
-            }
-
-            List<CodegenProperty> codegenProperties = new ArrayList<>();
-            if (model.getComposedSchemas() == null || (model.getComposedSchemas() != null && model.getComposedSchemas().getAllOf() != null)) {
-                // If the model is an allOf or does not have any composed schemas, then we can use the model's properties.
-                codegenProperties.addAll(model.vars);
-            } else {
-                // If the model is no model, but is a
-                // anyOf or oneOf, add all first level options
-                // from anyOf or oneOf.
-                codegenProperties.addAll(inheritedProperties);
```

Comment:
can you please add some comments similar to those in the code deleted/refactored above?